### PR TITLE
`General`: Make Exercise View Scrollable

### LINKED ISSUE #8674: Fix Problem Statements not getting displayed on mobile Phones
### Describe the bug

When opening a programming exercise view on a mobile device with a small screen width, the problem statement itself can not be seen but only the sidebar with the different exercises. 
The website is not scrollable to the side in mobile View neither in desktop view on the phone. Therefore the user can not scroll to the side to see the problem statement

### To Reproduce

1. Go to a course with programming exercises on your mobile phone
2. Click on a programming exercise
3. Try to scroll to the side to see the problem statement

### Expected behavior

As a user I either want to be able to minimize the sidebar on my mobile phone
Or be able to scroll to the side to see the problem statement

### Screenshots

https://github.com/ls1intum/Artemis/assets/95364200/4252803b-c7e8-4feb-b052-9d7a6726cd1a

### Which version of Artemis are you seeing the problem on?

7.1.0

### What browsers are you seeing the problem on?

Safari

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by zhngharry**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Fixes #8674 

The Course Overview page can't be properly displayed on some mobile devices with narrow screens (even with the sidebar collapsed). This PR implements a quick fix to make the view scrollable for those screens

### Description

Added CSS to make it scrollable

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis
2. Navigate to Course Overview
3. Take a look at Exercise Page with a narrow screen
4. Verify that everything is able to be displayed now

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### Screenshots

https://github.com/ls1intum/Artemis/assets/98264403/efee73e5-a49b-447f-a5eb-f27cc026394c

### CODE REVIEW

**Review by JohannesWt — APPROVED** [review]
Not sure if many people will use Artemis like this but if so - they can scroll now :) tested on TS3 - works as described

File: src/main/webapp/app/overview/course-overview.scss [hunk 1]
```diff
@@ -179,6 +179,12 @@ canvas#complete-chart {
     animation-iteration-count: initial;
 }
 
+@media (max-width: 500px) {
```
Comment (muradium) [inline src/main/webapp/app/overview/course-overview.scss:182]: Are we intentionally assuming here that sidebar is hidden? When sidebar is visible, overall width goes beyond `500px`.
Considering that sidebar takes around `300px`, I'd set `max-width: 800px`

Comment (zhngharry) [reply src/main/webapp/app/overview/course-overview.scss:182]: I originally wanted to encourage users to collapsing the sidebar, but 500px was inaccurate, Ive tested it again and the correct width-border is 655px if we want to make it scrollable even if the sidebar is open. Anything above the 655px can be displayed properly and is already being handled by shifting elements, and the scroll becomes redundant. I'd therefore propose using 655px as the new border.

**Review by rstief — APPROVED** [review]
code lgtm

**Review by az108 — APPROVED** [review]
LGTM

**Review by Jan-Thurner — APPROVED** [review]
Tested on TS1 for both regular exercises and Exams. As Exams don't have a sidebar (I only noticed it during testing myself), I don't think we need specific exam mode tests. It works as expected and the view is now scrollable.

**Review by sarpsahinalp — APPROVED** [review]
Tested locally works as expected

**Review by muradium — APPROVED** [review]
LGTM, thanks for the adjustments!

**Review by egekurt123 — APPROVED** [review]
Works as expected on ts3

**Comment by janthoXO** [comment]
https://github.com/ls1intum/Artemis/assets/95364200/25c6be0f-315d-496a-8901-84ccef6ecc92

This weeks FPV homework had the same problem that it was not properly displayed. This is the case on a notebook now as well. so it does not online concern mobile devices with a small width, but is a general problem

**Review by edkaya — APPROVED** [review]
code lgtm

**Comment by janthoXO** [comment]
@bassner or any maintainer. can you review this PR so it can be merged?

**Pull request merged**